frontend/javascripts/oxalis/shaders/main_data_fragment.glsl.ts

```diff
@@ -251,5 +284,249 @@ void main() {
+out vec4 worldCoord;
+flat out vec2 index;
+flat out vec3 flatVertexPos;
+flat out uvec4 outputCompressedEntry[<%= globalLayerCount %>];
+flat out uint outputMagIdx[<%= globalLayerCount %>];
+flat out uint outputSeed[<%= globalLayerCount %>];
+flat out float outputAddress[<%= globalLayerCount %>];
+varying vec4 modelCoord;
+varying vec2 vUv;
+varying mat4 savedModelMatrix;
+
+
+
+
+const int dataTextureCountPerLayer = <%= dataTextureCountPerLayer %>;
+uniform highp usampler2D lookup_texture;
+uniform highp uint lookup_seeds[3];
+uniform highp uint LOOKUP_CUCKOO_ENTRY_CAPACITY;
+uniform highp uint LOOKUP_CUCKOO_ELEMENTS_PER_ENTRY;
+uniform highp uint LOOKUP_CUCKOO_ELEMENTS_PER_TEXEL;
+uniform highp uint LOOKUP_CUCKOO_TWIDTH;
+
+<% _.each(colorLayerNames, function(name) { %>
+  uniform vec3 <%= name %>_color;
+  uniform float <%= name %>_min;
+  uniform float <%= name %>_max;
+  uniform float <%= name %>_is_inverted;
+<% }) %>
+
+<% _.each(layerNamesWithSegmentation, function(name) { %>
+  uniform sampler2D <%= name %>_textures[dataTextureCountPerLayer];
+  uniform float <%= name %>_data_texture_width;
+  uniform float <%= name %>_alpha;
+  uniform float <%= name %>_gammaCorrectionValue;
+  uniform float <%= name %>_unrenderable;
+  uniform mat4 <%= name %>_transform;
+  uniform bool <%= name %>_has_transform;
+<% }) %>
+
+uniform float activeMagIndices[<%= globalLayerCount %>];
+uniform uint availableLayerIndexToGlobalLayerIndex[<%= globalLayerCount %>];
+uniform vec3 resolutions[20];
+uniform int representativeLayerIdxForMag;
+
+<% if (hasSegmentation) { %>
+  // Custom color cuckoo table
+  uniform highp usampler2D custom_color_texture;
+  uniform highp uint custom_color_seeds[3];
+  uniform highp uint CUCKOO_ENTRY_CAPACITY;
+  uniform highp uint CUCKOO_ELEMENTS_PER_ENTRY;
+  uniform highp uint CUCKOO_ELEMENTS_PER_TEXEL;
+  uniform highp uint CUCKOO_TWIDTH;
+
+  uniform vec4 activeCellIdHigh;
+  uniform vec4 activeCellIdLow;
+  uniform bool isMouseInActiveViewport;
+  uniform bool showBrush;
+  uniform bool isProofreading;
+  uniform float segmentationPatternOpacity;
+
+  uniform bool isMappingEnabled;
+  uniform float mappingSize;
+  uniform bool hideUnmappedIds;
+  uniform sampler2D segmentation_mapping_texture;
+  uniform sampler2D segmentation_mapping_lookup_texture;
+<% } %>
+
+uniform float sphericalCapRadius;
+uniform float viewMode;
+uniform float alpha;
+uniform bool renderBucketIndices;
+uniform vec3 bboxMin;
+uniform vec3 bboxMax;
+uniform vec3 globalPosition;
+uniform vec3 activeSegmentPosition;
+uniform float zoomValue;
+uniform bool useBilinearFiltering;
+uniform vec3 globalMousePosition;
+uniform bool isMouseInCanvas;
+uniform float brushSizeInPixel;
+uniform float planeID;
+uniform vec3 addressSpaceDimensions;
+uniform vec4 hoveredSegmentIdLow;
+uniform vec4 hoveredSegmentIdHigh;
+
+const float bucketWidth = <%= bucketWidth %>;
+const float bucketSize = <%= bucketSize %>;
+const float l_texture_width = <%= l_texture_width %>;
+
```

Comment:
Does the above section mirror the whole variable declarations of the fragment shader?

Reply:
good point! I could dry most of it :)